packages/toolkit/src/query/defaultSerializeQueryArgs.ts

```diff
@@ -17,33 +18,40 @@ export const defaultSerializeQueryArgs: SerializeQueryArgs<any> = ({
   if (typeof cached === 'string') {
     serialized = cached
   } else {
-    const stringified = JSON.stringify(queryArgs, (key, value) =>
-      isPlainObject(value)
+    const stringified = JSON.stringify(queryArgs, (key, value) => {
+      // Use custom replacer first before applying key-sorting behavior:
+      value = replacer ? replacer(key, value) : value
+      // Sort the object keys before stringifying, to prevent useQuery({ a: 1, b: 2 }) having a different cache key than useQuery({ b: 2, a: 1 })
+      value = isPlainObject(value)
         ? Object.keys(value)
             .sort()
             .reduce<any>((acc, key) => {
               acc[key] = (value as any)[key]
               return acc
             }, {})
-        : value,
-    )
+        : value
+      return value
+    })
     if (isPlainObject(queryArgs)) {
       cache?.set(queryArgs, stringified)
     }
     serialized = stringified
   }
-  // Sort the object keys before stringifying, to prevent useQuery({ a: 1, b: 2 }) having a different cache key than useQuery({ b: 2, a: 1 })
   return `${endpointName}(${serialized})`
 }
 
 export type SerializeQueryArgs<QueryArgs, ReturnType = string> = (_: {
   queryArgs: QueryArgs
   endpointDefinition: EndpointDefinition<any, any, any, any>
   endpointName: string
+  // Allows for a custom stringify replacer while keeping key-sorting behavior. e.g. for serializing bigint.
+  replacer?: (key: string, value: any) => {}
```

Comment:
Special reviewer attention is requested here: I'm not sure I fully understand `SerializeQueryArgs` vs. `InternalSerializeQueryArgs` and whether it's proper to update both of them here.